src/python/alias.cpp

```diff
@@ -95,16 +95,16 @@ static nb::object variant_module(nb::handle variant) {
 
 /// Sets the variant
 static void set_variant(nb::args args) {
-    nb::object new_variant{};
-    for (auto arg : args) {
-        // Find the first valid & compiled variant in the arguments
+    nb::list valid_variants{};
+    for (const auto &arg : args) {
         if (PyDict_Contains(variant_modules, arg.ptr()) == 1) {
-            new_variant = nb::borrow(arg);
-            break;
+            // Variant is at least compiled, we can attempt to use it.
+            valid_variants.append(nb::borrow(arg));
```

Comment:
I'm not certain about the ref counting here.

Reply:
Fairly certain that the `borrow` is ok. Are there currently any leaks?
I can imagine that a failed `set_variant` itself might leak some stuff too actually.

Reply:
You should not need the borrow at all IMO.